build/commands/lib/config.js

```diff
@@ -221,6 +221,7 @@ const Config = function () {
   this.goma_offline = false
   this.use_libfuzzer = false
   this.brave_ai_chat_endpoint = getNPMConfig(['brave_ai_chat_endpoint']) || ''
+  this.brave_ai_chat_premium_endpoint = getNPMConfig(['brave_ai_chat_premium_endpoint']) || ''
```

Comment:
We won't need to add these buildflags as once https://github.com/brave/brave-core/pull/20249 is merged we can just prepend to the `brave_services_production_domain` build flag